internal/handler/handler_test.go

```diff
@@ -0,0 +1,746 @@
+	id := uuid.New()
+	//revive:disable
+	ctx = context.WithValue(ctx, "correlation_id", id.String()) //nolint:staticcheck
+	//revive:enable
+	r = r.WithContext(ctx)
+	return r
+}
+
+func TestHand_CreateEmployee(t *testing.T) {
+	initTest()
+	firstPosition := internal.Position{ID: createPosID(), Salary: decimal.New(500, 0), Name: "worker"}
+	repos.AddPosition(&firstPosition)
+	id, err := uuid.Parse(positionIDs[0])
+	if err != nil {
+		t.Fatalf("Error: %v ", err)
+	}
+	firstEmployee := internal.Employee{ID: createEmpID(), FirstName: "Bob", LasName: "Vik", PositionID: id}
+	secondEmployee := internal.Employee{ID: createEmpID(), FirstName: "", LasName: "Vik", PositionID: id}
+	thirdEmployee := internal.Employee{ID: createEmpID(), FirstName: "Fox", LasName: "Fok", PositionID: uuid.New()}
+	jsonFirstEmployee, _ := json.Marshal(firstEmployee)
+	jsonSecondEmployee, _ := json.Marshal(secondEmployee)
+	jsonThirdEmployee, _ := json.Marshal(thirdEmployee)
+	reader := strings.NewReader(string(jsonFirstEmployee))
+	testTable := []struct {
+		URL      string
+		method   string
+		expected int
+		read     io.Reader
+	}{
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 200,
+			read:     reader,
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 500,
+			read:     strings.NewReader("s"),
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonThirdEmployee)),
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonSecondEmployee)),
+		},
+	}
+	for _, testCase := range testTable {
+		r, err := http.NewRequest("POST", "http://localhost:8080/employee", testCase.read)
+		r.Header.Set("content-type", "application/json")
+		r = createTestContext(r)
+		if err != nil {
+			t.Fatalf("could not created request: %v", err)
+		}
+		w := httptest.NewRecorder()
+		handler.CreateEmployee(w, r)
+		result := w.Result()
+		defer result.Body.Close()
+		if result.StatusCode != testCase.expected {
+			t.Fatalf("expected status created; get %v", result.StatusCode)
+		}
+	}
+}
+
+func TestHand_CreatePosition(t *testing.T) {
+	initTest()
+	firstPosition := internal.Position{Salary: decimal.New(500, 0), Name: "worker"}
+	fakePosition := internal.Employee{ID: createEmpID(), FirstName: "V", LasName: "T", PositionID: uuid.New()}
+	jsonFirstPosition, _ := json.Marshal(firstPosition)
+	jsonFakePosition, _ := json.Marshal(fakePosition)
+	reader := strings.NewReader(string(jsonFirstPosition))
+	testTable := []struct {
+		URL      string
+		method   string
+		expected int
+		read     io.Reader
+	}{
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 500,
+			read:     strings.NewReader("s"),
+		},
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 200,
```

Comment:
According to REST if entity created successfully you have to return 201 code

Reply:
Sorry, i agree with you.